core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -216,28 +216,103 @@ func (ec *ethConfirmer) SetBroadcastBeforeBlockNum(blockNum int64) error {
 }
 
 func (ec *ethConfirmer) CheckForReceipts(ctx context.Context, blockNum int64) error {
-	batchSize := int(ec.config.EthRPCDefaultBatchSize())
-
 	attempts, err := ec.findEthTxAttemptsRequiringReceiptFetch()
+
 	if err != nil {
 		return errors.Wrap(err, "findEthTxAttemptsRequiringReceiptFetch failed")
 	}
 	if len(attempts) == 0 {
 		return nil
 	}
-	if batchSize == 0 {
-		batchSize = len(attempts)
-	}
 
 	logger.Debugw(fmt.Sprintf("EthConfirmer: fetching receipts for %v transaction attempts", len(attempts)), "blockNum", blockNum)
 
+	attemptsByAddress := make(map[gethCommon.Address][]models.EthTxAttempt)
+	for _, att := range attempts {
+		attemptsByAddress[att.EthTx.FromAddress] = append(attemptsByAddress[att.EthTx.FromAddress], att)
+	}
+
+	for from, attempts := range attemptsByAddress {
+		ctxInner, cancel := eth.DefaultQueryCtx(ctx)
+		latestBlockNonce, err := ec.getNonceForLatestBlock(ctxInner, from)
```

Comment:
Could we save time by batch fetching it before the loop?

Reply:
hmm might be a bit of overkill, because `att.EthTx.FromAddress` can have just max two possible values? What do you think?

Reply:
Most node operators only run with one key, so you are probably right. I was imagining some future-proofing if we support multiple keys, but its not that important.